### PR TITLE
Add mp2p_icp for indexing

### PR DESCRIPTION
**Pull request by jlblancoc**
# Please Add This Package to be indexed in the rosdistro.

iron

# The source is here:

https://github.com/MOLAorg/mp2p_icp

# Checks
 - [X] All packages have a declared license in the package.xml
 - [X] This repository has a LICENSE file
 - [X] This package is expected to build on the submitted rosdistro

NOTE: This package is right now already published within another package ("mola") but in the next release it will be released as an independently package & git repository, hence this PR.

### CODE REVIEW

**Comment by audrow** [comment]
Same as https://github.com/ros/rosdistro/pull/39210#issuecomment-1856575288.

**Comment by jlblancoc** [comment]
Comments solved in original thread; plus `<license file=...>` attribute [added upstream](https://github.com/MOLAorg/mp2p_icp/commit/a487073e0053738fbcd483ca393c57c275ebe2b5).

File: iron/distribution.yaml [hunk 1]
```diff
@@ -3181,6 +3181,16 @@ repositories:
       url: https://github.com/ros-planning/moveit_visual_tools.git
       version: ros2
     status: maintained
+  mp2p_icp:
```
Comment (clalancette) [inline iron/distribution.yaml:3184]: I have a question on how this package is going to interact with https://github.com/ros/rosdistro/blob/5bb368ac3182c13773638f8a9fee63ad1202c1c1/iron/distribution.yaml#L3078 , which is already released.  As far as I understand, this new package has the same name as that package, so you won't be able to release it here.  What is your plan for dealing with this?

Comment (jlblancoc) [reply iron/distribution.yaml:3184]: Hi @clalancette !

Yes, you are correct. Long story short: 
- Current state: one git repo (with submodules) named "mola".
- New state: 3 git repos (no submodules anymore)
  - mola_common : Already merged for indexing in #39230  (Read the "NOTE" section at the end of that PR).
  - mp2p_icp: This PR
  - mola: New releases to be made, now of course without the other two packages above, once they are merged and bloom-released.

If it's an issue to have the same package released twice, even during 1 day, I can create a PR to remove the "release:" section of "mola". 

Thanks, and sorry for the mess, this new state of things makes more sense thinking of the modularity of new forthcoming packages.

**Comment by jlblancoc** [comment]
I'll reopen this one after #39311 to avoid problems.

**Pull request closed without merging**